File: Lua/Exercício 2.lua
local lista = {}
local entrada
local quantidade = 0

print("Digite números (digite 0 para parar): ")

while true do
    entrada = tonumber(io.read())
    if entrada == 0 then -- condição de parada
        break
    end
    table.insert(lista, entrada)
    quantidade = quantidade + 1
end

local maximo = lista[1] -- assume que o primeiro é o maior
for indice = 2, quantidade do
    if lista[indice] > maximo then
        maximo = lista[indice]
    end
end

print("O maior número é: " .. maximo)

File: Go/Exercício 2.go
package main
import "fmt"

func main() {
	var entrada int

	fmt.Print("Digite um número: ")
	fmt.Scanln(&entrada)

	restoDivisao := entrada % 2
	
	switch restoDivisao {
	case 0:
		fmt.Println("O valor informado é par")
	default:
		fmt.Println("O valor informado é ímpar")
	}
}
